astropy/time/core.py

```diff
@@ -225,12 +224,19 @@ def __init__(self, val, val2=None, format=None, scale=None,
                                  'one for each time.'
                                  .format(self.location.shape, self.shape))
 
-    def _init_from_vals(self, val, val2, format, scale, copy):
+    def _init_from_vals(self, val, val2, format, scale, copy,
+                        precision=None, in_subfmt=None, out_subfmt=None):
         """
         Set the internal _format, scale, and _time attrs from user
         inputs.  This handles coercion into the correct shapes and
         some basic input validation.
         """
+        if precision is None:
```

Comment:
I had the same comment to myself during development and spent a little time thinking about it.

You can't have defaults in `__init__` because of the new lines 200-209 above.  This would clobber existing values when creating a Time object from another time object.
```
In [2]: t = Time.now()

In [3]: t.precision = 5

In [4]: t2 = Time(t)

In [5]: t2
Out[5]: <Time object: scale='utc' format='datetime' value=2016-01-04 19:11:17.006990>
```
If you have `precision=3, in_subfmt='*', out_subfmt='*'`` in the `_init_from_vals` signature (or use required positional args), then there are problems because the calling routine will currently pass `None` for those values.  Also with required positional args then one would need to repeat or make up defaults in `TimeDelta`.

Anyway, I see your point, and maybe there is a solution but it isn't quite as easy as you said.  Basically what we need for code simplicity here is just using `**kwargs`, but I don't like that for API.